lib/package/plugins/TD004-targetSslInfo.js

```diff
@@ -39,52 +39,77 @@ const onTargetEndpoint =
+                messages.push('SSLInfo configuration is not Enabled');
+              }
+
+              elts = htc.select(`SSLInfo/IgnoreValidationErrors`);
+              let ignoreErrors = elts && elts[0] && elts[0].childNodes && elts[0].childNodes[0] &&
+                elts[0].childNodes[0].nodeValue == 'true';
+              if (ignoreErrors) {
+                messages.push('SSLInfo configuration includes IgnoreValidationErrors = true');
+              }
+
+              elts = htc.select(`SSLInfo/TrustStore`);
+              let hasTrustStore = elts && (elts.length == 1) && elts[0].childNodes && elts[0].childNodes[0];
+              if ( ! hasTrustStore) {
+                messages.push('Missing TrustStore in SSLInfo');
+              }
+
+              elts = htc.select(`SSLInfo/ClientAuthEnabled`);
+              let hasClientAuth = elts && (elts.length == 1) && elts[0].childNodes && elts[0].childNodes[0] &&
+                elts[0].childNodes[0].nodeValue == 'true';
+
+              // check for keystore, keyalias
+              elts = htc.select(`SSLInfo/KeyStore`);
+              let hasKeyStore = elts && (elts.length == 1) && elts[0].childNodes && elts[0].childNodes[0];
+              elts = htc.select(`SSLInfo/KeyAlias`);
+              let hasKeyAlias = elts && (elts.length == 1) && elts[0].childNodes && elts[0].childNodes[0];
+
+              if ( hasClientAuth && (! hasKeyStore || ! hasKeyAlias)) {
+                messages.push('When ClientAuthEnabled = true, use a KeyStore and KeyAlias');
+              }
+              if ( ! hasClientAuth && (hasKeyStore || hasKeyAlias)) {
+                messages.push('When ClientAuthEnabled = false, do not use a KeyStore and KeyAlias');
+              }
+            }
           }
-
-          elts = htc.select(`SSLInfo/ClientAuthEnabled`);
-          let hasClientAuth = elts && (elts.length == 1) && elts[0].childNodes && elts[0].childNodes[0] &&
-            elts[0].childNodes[0] == 'true';
-
-          // check for keystore, keyalias
-          elts = htc.select(`SSLInfo/KeyStore`);
-          let hasKeyStore = elts && (elts.length == 1) && elts[0].childNodes && elts[0].childNodes[0];
-          elts = htc.select(`SSLInfo/KeyAlias`);
-          let hasKeyAlias = elts && (elts.length == 1) && elts[0].childNodes && elts[0].childNodes[0];
-
-          if ( hasClientAuth && (! hasKeyStore || ! hasKeyAlias)) {
-            messages.push('When ClientAuthEnabled = true, use a KeyStore and KeyAlias');
+          else if (sslInfos.length == 1) {
+              messages.push(`SSLInfo should not be used with an insecure http url`);
           }
-          if ( ! hasClientAuth && (hasKeyStore || hasKeyAlias)) {
-            messages.push('When ClientAuthEnabled = false, do not use a KeyStore and KeyAlias');
+        }
+        else if (loadBalancers.length == 1) {
+          if (sslInfos.length == 1) {
+            messages.push(`HTTPTargetConnection/SSLInfo should not be used with LoadBalancer`);
```

Comment:
@DinoChiesa - should we add more info that it should configured as part of target server config?

Reply:
Yes, how about `With LoadBalancer, SSLInfo should be configured within TargetServer, not under HTTPTargetConnection` ?